File: partA3.java
public class partA3 {
    public static void main(String[] args) {
        int faces = 6;

        int[] sumFreq = new int[13]; 
        for (int i = 1; i <= faces; i++) {
            for (int j = 1; j <= faces; j++) {
                int sum = i + j;
                sumFreq[sum]++;
            }
        }
        int totalC = faces * faces;

        System.out.println("Probability of all possible sums occurring:");

        for (int sumValue = 2; sumValue <= 12; sumValue++) {
            double prob = (double) sumFreq[sumValue] / totalC;
            String y = String.format("%.2f", prob);
            System.out.println("Sum: " + sumValue + ", Probability: " + y);
        }
    }
}

File: partA3.py

faces = 6
sum_freq = {}
for i in range(2, 13):
    sum_freq[i] = 0

for i in range(1, faces + 1):
    for j in range(1, faces + 1):
        sum_freq[i + j] += 1

total_combinations = faces ** 2

print("Probability of all possible sums occurring:")
for sum_val in range(2, 13):
    probability = sum_freq[sum_val] / total_combinations
    print(f"Sum: {sum_val}, Probability: {probability:.2f}")
